src/java/us/kbase/templates/module_typespec.vm.properties

```diff
@@ -84,8 +84,15 @@ module ${module_name} {
     funcdef count_contigs(workspace_name,contigset_id) returns (CountContigsResults) authentication required;
 #end
 #else
+    typedef structure {
+        string report_name;
+        string report_ref;
+    } ReportResults;
+
     /*
-        Insert your typespec information here.
+        This example function takes accepts any number of parameters and returns results in a KBaseReport
     */
+    funcdef run_${module_name}(UnspecifiedObject params) returns (ReportResults output) authentication required;
```

Comment:
Why not make this a mapping<string, UnspecifiedObject> instead? At least then the param names are defined as strings.

Reply:
Sure

Reply:
$params is a reference to a hash where the key is a string and the value is an UnspecifiedObject, which can hold any non-null object

I do believe that sometimes we get null parameters back so this might prove an issue

Reply:
Bigger problem, can't use the builder pattern `.withMinLength(-10)` with either form of UnspecifiedObject as the Java example demonstrates.

Reply:
UObjects can take nulls, so I think that info is out of date:

```
	public static void main(String[] args) throws Exception {
		final ObjectMapper m = new ObjectMapper().registerModule(new JacksonTupleModule());
		
		final String json = m.writeValueAsString(ImmutableMap.of("key", new UObject(null)));
		System.out.println(json);
		
		final Map<String, UObject> instantiated =
				m.readValue(json, new TypeReference<Map<String, UObject>>() {});
		System.out.println(instantiated);
	}
```

Reply:
> Bigger problem, can't use the builder pattern .withMinLength(-10) with either form of UnspecifiedObject as the Java example demonstrates.

Yep, the compiler can't compile the fluent setters (not a builder btw, just setters) if it doesn't know the key names and types.